src/model/SplineSegment.h

```diff
@@ -0,0 +1,114 @@
+/*
+ * Xournal++
+ *
+ * A spline segment
+ *
+ * @author Xournal++ Team
+ * https://github.com/xournalpp/xournalpp
+ *
+ * @license GNU GPLv2 or later
+ */
+
+#pragma once
+
+#include <gtk/gtk.h>
+
+#include "model/Element.h"
+#include "model/Stroke.h"
+
+#include "Point.h"
+
+
+/**
+ * @brief A class to handle splines segments
+ *
+ * Every spline segment is defined by two knot point and two control points.
+ * This contrasts to the cairo-draw command cairo_curve_to, where everything is relative to the current position of
+ * another knot,
+ */
+
+
+class SplineSegment {
+public:
+    SplineSegment() = default;
+    ~SplineSegment() = default;
+
+    /**
+     * @brief A linear spline segment from two points
+     * @param p the first knot
+     * @param q the second knot
+     */
+    SplineSegment(const Point& p, const Point& q);
+
+    /**
+     * @brief A spline segment from four points
+     * @param p the first knot
+     * @param fp the first control point
+     * @param sp the second control point
+     * @param q the second knot
+     */
+    SplineSegment(const Point& p, const Point& fp, const Point& sp, const Point& q);
+
+    /**
+     * @brief draw the spline segment via the cairo_curve_to command
+     * @param cr the cairo context on which the spline segment ought to be drawn
+     */
+    void draw(cairo_t* cr);
+
+    /**
+     * @brief Convert the spline segment to a vector of points.
+     * @return A point list which represents the spline segment without the start point.
+     */
+    std::vector<Point> toVectorOfPoints();
```

Comment:
Here the declaration of my suggestion from above:
Also I did not understand, why the first point is missing.

```suggestion
    /**
     * @brief Convert the spline segment to a vector of points.
     * @return A point list which represents the spline segment.
     */
    std::list<Point> toPointSequence() const;

    /**
     * @brief Convert the spline segment to a vector of points.
     * @return A point list which represents the spline segment without the end point.
     */
    auto toPointSequenceImpl() const -> std::list<Point>;
```

Reply:
That "without the start point" comment was a mistake (I planned it differently earlier).